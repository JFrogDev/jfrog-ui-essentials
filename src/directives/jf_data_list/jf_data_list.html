<ul class="data-list">
	<li class="data-list-row" ng-repeat="item in jfDataList.formattedItems" ng-if="!item.isHidden">
		<div class="data-list-item data-label">{{item.label}}:</div>
		<div class="data-list-item data-value">{{item.value}}</div> <!-- TODO: copy button support (ng-if="item.copy") -->
	</li>
</ul>