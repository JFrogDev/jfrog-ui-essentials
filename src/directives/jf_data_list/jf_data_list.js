class jfDataListController {
	constructor($scope) {
		this.$scope = $scope;

		this.$scope.$watch('jfDataList.items', items => {
			if (items) {
				this.formattedItems = _.filter(items, (item) => {
					return item.value != '';
				});
			}
		})
	}
}


export function jfDataList() {
	return {
		restrict: 'E',
		scope: {
			items: '='
		},
		controller: jfDataListController,
		controllerAs: 'jfDataList',
		bindToController: true,
		templateUrl: 'directives/jf_data_list/jf_data_list.html'
	}
}