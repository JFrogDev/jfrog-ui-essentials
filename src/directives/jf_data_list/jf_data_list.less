.data-list {
  width: 100%;
  table-layout: fixed;
  &-item {
    &-label {
      line-height: 1.3em;
      padding-right: 20px;
      width: 25%;
      font-weight: 600;
    }
    &-value {
      line-height: 2.5em;
      .value {
        display: flex;
        width: 100%;
        & > :first-child {
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
        }
        .copy {
          margin-left: 8px;
        }
        .tag {
          &:first-child {
            margin-left: 0;
          }
          display: inline-block;
          background: transparent;
          border-radius: 3px;
          padding: 0 8px;
          margin: 6px 0 0 8px;
          height: 26px;
          line-height: 24px;
          color: @jfColorAlmostBlack;
          border: 1px solid @jfColorAlmostBlack;
          &.overflowing-child{
            display: none;
          }
        }
        .gridcell-showall{
          display: inline-block;
          position: relative;
          right: 0;
          padding: 0 10px;
          outline: none;
          white-space: nowrap;
        }
      }
    }
  }
}