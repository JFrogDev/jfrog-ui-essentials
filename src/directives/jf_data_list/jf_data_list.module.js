import jfDataListCompiledItem from './components/jf_data_list_item/jf_data_list_compiled_item';
import jfDataListItem from './components/jf_data_list_item/jf_data_list_item';
import jfDataList from './jf_data_list';

export const MODULE_DATA_LIST = 'jf.data.list';
angular.module(MODULE_DATA_LIST, [])
       .directive('jfDataListCompiledItem', jfDataListCompiledItem)
       .directive('jfDataListItem', jfDataListItem)
       .directive('jfDataList', jfDataList);