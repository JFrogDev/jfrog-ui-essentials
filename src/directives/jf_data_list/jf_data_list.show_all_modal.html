<div id="show-all-modal">
    <div class="modal-header">
        <button type="button" class="close" data-dismiss="close" aria-label="Close" @click="$dismiss()"><span aria-hidden="_true">Ã—</span></button>
        <h4 class="modal-title" id="popup-header">{{items.length}} {{colName}} For {{objectName}} '{{rowName}}'</h4>
    </div>
    <div class="modal-body clearfix">
        <input type="text" name="filter" class="input-text" v-model="filter.text" placeholder="Filter">


        <div class="group-list-wrapper" v-if="!noResults()">
            <ul class="group-list">
                <li class="group-list-item" v-jf-tooltip-on-overflow="''" v-for="_(item, $index) in items" v-if="filterItem(item)">
                    <div v-if="!item.url" v-html="item.label"></div>
                    <a v-if="item.url" class="jf-link" :href="item.url" v-html="item.label" target="_blank"></a>
                </li>
            </ul>
        </div>

        <div class="empty-filter-placeholder filter-no-results" v-if="noResults()">
            Current filter has no results. <a href="" class="jf-link" @click.prevent="filter.text = ''">Clear filter</a>
        </div>

    </div>
    <div class="modal-footer">
        <button class="btn btn-default" @click="$dismiss()" id="popup-cancel">Close</button>
    </div>
</div>