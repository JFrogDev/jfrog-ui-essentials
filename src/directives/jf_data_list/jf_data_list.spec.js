import {items} from './jf_data_list.mock';

describe('unit test: jf_table_view directive & JFTableViewOptions service', () => {

    var $scope;
    var $rootScope;
    var $q;
    var $timeout;
    let $elem;

    const setup = (_$timeout_, _$q_, _$rootScope_) => {
        $timeout = _$timeout_;
        $rootScope = _$rootScope_;
        $q = _$q_;
    }

    const compileDirective = (attr) => {
        ({$scope, $elem} = window.compileDirectiveAndGetElement('jf-data-list', attr));
    }

    beforeEach(angular.mock.module('jfrog.ui.essentials'));
    beforeEach(inject(setup));

    beforeEach(() => {
        compileDirective({
            items,
        });
    });

    it('should $elem be compiled', function () {
        expect($elem.html().length).toBeGreaterThan(0);
    });

});