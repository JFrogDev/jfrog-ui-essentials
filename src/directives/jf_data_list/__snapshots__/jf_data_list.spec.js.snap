// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test: jf_table_view directive & JFTableViewOptions service should match snapshot 1`] = `
"<table class=\\"data-list\\">
	<tbody><!-- ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Plain text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">More plain text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Text with copy to clipboard:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) --><div class=\\"copy ng-scope\\" ng-if=\\"item.copy &amp;&amp; !jfDataList.isArray(item.value)\\">
					<jf-clip-copy text-to-copy=\\"item.value\\" class=\\"ng-isolate-scope\\"><span class=\\"clip-copy-button\\">
    <a ng-show=\\"!jfClipCopy.isSafari &amp;&amp; !!jfClipCopy.textToCopy\\" class=\\"copy-to-clip icon icon-copy-to-clipboard-2\\" data-clipboard-text=\\"{{jfClipCopy.textToCopy}}\\" jf-tooltip=\\"{{jfClipCopy.tooltipText}}\\"></a>
    <a ng-show=\\"jfClipCopy.isSafari &amp;&amp; !!jfClipCopy.textToCopy\\" class=\\"copy-to-clip icon icon-copy-to-clipboard\\" clip-copy=\\"jfClipCopy.textToCopy\\" clip-click=\\"jfClipCopy.onSuccessfulCopy()\\" jf-tooltip=\\"{{jfClipCopy.tooltipText}}\\"></a>
</span>

</jf-clip-copy>
				</div><!-- end ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Overflowing text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Overflowing text with copy:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) --><div class=\\"copy ng-scope\\" ng-if=\\"item.copy &amp;&amp; !jfDataList.isArray(item.value)\\">
					<jf-clip-copy text-to-copy=\\"item.value\\"></jf-clip-copy>
				</div><!-- end ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Empty text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">List with some clickable items:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">List with overflowing items:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Value is url:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) --><a ng-if=\\"item.isUrl &amp;&amp; item.value &amp;&amp; item.value.length &amp;&amp; ((!item.hasOwnProperty('url')) || (item.url &amp;&amp; item.url.length))\\" ng-href=\\"http://www.lorem_ipsum_dolor_sit_amet_consectetur_adipisicing_elit.com\\" ng-bind-html=\\"item.value\\" target=\\"_blank\\" class=\\"jf-link ng-binding ng-scope ng-isolate-scope\\" jf-tooltip-on-overflow=\\"\\" href=\\"http://www.lorem_ipsum_dolor_sit_amet_consectetur_adipisicing_elit.com\\">http://www.lorem_ipsum_dolor_sit_amet_consectetur_adipisicing_elit.com</a><!-- end ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Value with separate url:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) --><a ng-if=\\"item.isUrl &amp;&amp; item.value &amp;&amp; item.value.length &amp;&amp; ((!item.hasOwnProperty('url')) || (item.url &amp;&amp; item.url.length))\\" ng-href=\\"https://www.lipsum.com/\\" ng-bind-html=\\"item.value\\" target=\\"_blank\\" class=\\"jf-link ng-binding ng-scope ng-isolate-scope\\" jf-tooltip-on-overflow=\\"\\" href=\\"https://www.lipsum.com/\\">Link to lorem ipsum website</a><!-- end ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Empty url:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems --><!-- ngIf: !item.isHidden --><tr class=\\"data-list-item ng-scope\\" ng-repeat=\\"item in jfDataList.formattedItems\\" ng-if=\\"!item.isHidden\\">
		<td class=\\"data-list-item-label ng-binding\\">Empty url &amp; text:</td>
		<td class=\\"data-list-item-value\\">
			<div class=\\"value\\">
				<!-- ngIf: item.isUrl && item.value && item.value.length && ((!item.hasOwnProperty('url')) || (item.url && item.url.length)) -->
				<!-- ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) --><div ng-if=\\"!jfDataList.isArray(item.value) &amp;&amp; (!item.isUrl || (item.isUrl &amp;&amp; ((item.hasOwnProperty('url') &amp;&amp; !item.url.length) || (!item.value || !item.value.length))))\\" ng-bind-html=\\"jfDataList.getItemValue(item.value)\\" jf-tooltip-on-overflow=\\"\\" class=\\"ng-binding ng-scope ng-isolate-scope\\"></div><!-- end ngIf: !jfDataList.isArray(item.value) && (!item.isUrl || (item.isUrl && ((item.hasOwnProperty('url') && !item.url.length) || (!item.value || !item.value.length)))) -->
				<!-- ngIf: jfDataList.isArray(item.value) -->
				<!-- ngIf: item.copy && !jfDataList.isArray(item.value) -->
			</div>
		</td>
	</tr><!-- end ngIf: !item.isHidden --><!-- end ngRepeat: item in jfDataList.formattedItems -->
</tbody></table>"
`;
